### ISSUE TITLE
Server WFS-T does not respect srsName

### ISSUE THREAD

**Issue by elpaso** [qgis/QGIS#39371]
I might be doing something very silly or there or there is something very wrong in current WFS-T server implementation: 

## Step 1: project preparation
- create a shapefile EPSG:3044 and add it to a new project
- set project CRS to EPSG:3044
- in QGIS desktop create a feature in a known place for reference
- save the project and swich on WFS in server settings
- close QGIS desktop

## Step 2: qgis server
- serve the project created above

## Step 3: edit the layer with WFS-T in QGIS desktop
- open QGIS desktop
- load the WFS layer created above from QGIS server
- set current CRS to 4326
- start editing, create a feature
- commit changes

This is what happen behind the scenes:

- a transaction like the following is created (forget about 1.1.0: 1.0.0 has the same behavior):
```xml
<Transaction xmlns="http://www.opengis.net/wfs" service="WFS"
    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://www.qgis.org/gml http://localhost:8000/?SERVICE=WFS&amp;REQUEST=DescribeFeatureType&amp;VERSION=1.0.0&amp;TYPENAME=inverted_axis_3044"
    xmlns:gml="http://www.opengis.net/gml" version="1.1.0">
    <Insert xmlns="http://www.opengis.net/wfs">
        <inverted_axis_3044 xmlns="http://www.qgis.org/gml">
            <id xmlns="http://www.qgis.org/gml">2</id>
            <name xmlns="http://www.qgis.org/gml">due</name>
            <geometry xmlns="http://www.qgis.org/gml">
                <gml:MultiPolygon srsName="EPSG:4326">
                    <gml:polygonMember>
                        <gml:Polygon>
                            <gml:outerBoundaryIs>
                                <gml:LinearRing>
                                    <gml:coordinates cs="," ts=" ">11.09791694153736685,46.07951608016555411 11.09795489562553605,46.07078384865686616 11.1122882780524268,46.07123881425378187 11.11361274255637532,46.07886725543162498 11.09791694153736685,46.07951608016555411</gml:coordinates>
                                </gml:LinearRing>
                            </gml:outerBoundaryIs>
                        </gml:Polygon>
                    </gml:polygonMember>
                </gml:MultiPolygon>
            </geometry>
        </inverted_axis_3044>
    </Insert>
</Transaction>
```

Note that:
- `srsName` CRS is 4326 
- coordinates are also in 4326, so far so good

but apparently the server simply ignores  `srsName`.... and inserts a 4326 polygon into a 3044 dataset.

**Comment by elpaso** [comment]
@rldhont @pblottiere maybe I'm too tired and having bad dreams after too much bugfixing: can you confirm that the bug is in the server? I mean: it should honor `srsName`, doesn't it?

**Comment by rouault** [comment]
> I mean: it should honor srsName, doesn't it?

For WFS 1.1, according to http://docs.opengeospatial.org/is/04-094r1/04-094r1.html , the answer is yes (interesting part is the last 2 sentences):
"""The optional srsName attribute [on Insert] is used to assert the SRS of the incoming feature data, which can be useful if the incoming feature data does not have the SRS declared for each geometry. If the srsName attribute exists on an <Insert> element, its value must be equivalent to the value of <DefaultSRS> or any of the <OtherSRS> of the relevant feature types. If, however, the SRS is not supported, the WFS shall raise an exception as described in subclause 7.7. If the srsName is not specified on the <Insert> element, the WFS shall interpret that the feature data is given in the <DefaultSRS>, except where an SRS is specified on an included feature geometry. In this case, if the SRS for such a geometry is one of the <DefaultSRS> or <OtherSRS> values for the respective feature types, it will be transformed as required before insertion."""

**Comment by elpaso** [comment]
@rouault thanks for the clarification, I made a PR #39373 that fix both 1.1 and 1.0 , do you think I should limit the patch to 1.1? It seems logical to me that when srsName is specified we **always** need to transform the coordinates if they are different from the data source.
